petastorm/reader.py

```diff
@@ -603,11 +620,14 @@ def __iter__(self):
         return self
 
     def __next__(self):
-        try:
-            return self._results_queue_reader.read_next(self._workers_pool, self.schema, self.ngram)
-        except StopIteration:
-            self.last_row_consumed = True
-            raise
+        with self._statistics_collector.time_row_retrieval():
+            try:
+                results = self._results_queue_reader.read_next(self._workers_pool, self.schema, self.ngram)
+                self._statistics_collector.count_rows_retrieved()
```

Comment:
The name is a little misleading for the batch case, as it would count the batches. 

Would it make sense to blend `count_rows_retrieved` with `time_row_retrieval`? For example Petastorm could provide a 'performance counter' implementation (e.g. aggregate, number of measures, average/last/median an event took, etc). A customization piece should be just how the reporting is done.

Reply:
Yeah I was thinking about the name in the batch case.

And I think that is something we can provide

Reply:
So median is difficult to do since in order to calculate the median you need to store all values seen (could run into memory issues potentially). We could calculate means but not sure how much value that would add.

Also im trying to think of better names and nothing is obvious to me other than simply calling it 'next calls' or something like that.